BUG: GC fails to run when files other than metarange and ranges are found in _lakefs folder

When non range files exist in the path we get the following error:

```
Job failed, please check complete logs in configured logging destination. ExitCode: 1. Last few exceptions: Caused by: io.treeverse.jpebble.BadFileFormatException: Bad magic 39 38 64 31 39 22 0a 7d: wrong bytes Caused by: org.apache.spark.SparkException: Job aborted due to stage failure:
```

This issue becomes even more acute since our  ref dump documentation and procedure saves the ref dumps file under the `_lakefs` path

--- a/clients/spark/build.sbt
+++ b/clients/spark/build.sbt
@@ -1,4 +1,4 @@
-lazy val projectVersion = "0.14.2"
+lazy val projectVersion = "0.14.3"
 version := projectVersion
 lazy val hadoopVersion = "3.2.1"
 ThisBuild / isSnapshot := false

--- a/clients/spark/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala
+++ b/clients/spark/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala
@@ -99,8 +99,6 @@ class EntryRecordReader[Proto <: GeneratedMessage with scalapb.Message[Proto]](
 
     val gravelerSplit = split.asInstanceOf[GravelerSplit]
 
-    logger.info(s"Processing file: ${gravelerSplit.path}")
-
     val fs = gravelerSplit.path.getFileSystem(context.getConfiguration)
     fs.copyToLocalFile(false, gravelerSplit.path, new Path(localFile.getAbsolutePath), true)
     // TODO(johnnyaug) should we cache this?

--- a/clients/spark/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala
+++ b/clients/spark/src/main/scala/io/treeverse/clients/LakeFSInputFormat.scala
@@ -99,16 +99,27 @@ class EntryRecordReader[Proto <: GeneratedMessage with scalapb.Message[Proto]](
 
     val gravelerSplit = split.asInstanceOf[GravelerSplit]
 
+    logger.info(s"Processing file: ${gravelerSplit.path}")
+
     val fs = gravelerSplit.path.getFileSystem(context.getConfiguration)
     fs.copyToLocalFile(false, gravelerSplit.path, new Path(localFile.getAbsolutePath), true)
     // TODO(johnnyaug) should we cache this?
     sstableReader = new SSTableReader(localFile.getAbsolutePath, companion, true)
     if (!gravelerSplit.isValidated) {
       // this file may not be a valid range file, validate it
-      val props = sstableReader.getProperties
-      logger.debug(s"Props: $props")
-      if (new String(props("type")) != "ranges" || props.contains("entity")) {
-        return
+      try {
+        val props = sstableReader.getProperties
+        logger.debug(s"Props: $props")
+        if (new String(props("type")) != "ranges" || props.contains("entity")) {
+          return
+        }
+      } catch {
+        case e: io.treeverse.jpebble.BadFileFormatException =>
+          logger.error(s"Failed to read sstable, bad file format: ${gravelerSplit.path}", e)
+          throw new io.treeverse.jpebble.BadFileFormatException(
+            s"Bad file format in ${gravelerSplit.path}: ${e.getMessage}",
+            e
+          )
       }
     }
     rangeID = gravelerSplit.rangeID
@@ -259,8 +270,8 @@ class LakeFSAllRangesInputFormat extends LakeFSBaseInputFormat {
     while (it.hasNext) {
       val file = it.next()
       breakable {
-        if (file.getPath.getName == DummyFileName) {
-          logger.debug(s"Skipping dummy file ${file.getPath}")
+        if (file.getPath.getName == DummyFileName || file.getPath.getName.endsWith(".json")) {
+          logger.debug(s"Skipping file ${file.getPath}")
           break
         }
         splits += new GravelerSplit(

